src/Workspaces/Core/Portable/Rename/Renamer.RenameSymbolDocumentAction.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    public static partial class Renamer
+    {
+        /// <summary>
+        /// Action that will rename a type to match the current document name
+        /// </summary>
+        internal class RenameSymbolDocumentAction : RenameDocumentAction
```

Comment:
Would these types ever be public if people need to differentiate one from the other? We had the enum in the past and honestly your inheritence model is a lot nicer; wasn't sure if that might be a plan at some point.

Reply:
I think they _could_ be public, but I don't see a strong use case for that yet.

Reply:
sealed.

Reply:
Do we seal internal classes?

Reply:
we're starting to now.  (would be good to do everything).  it makes it explicit when you want to allow inheritance.